monitoring/boot_config_linux.go

```diff
@@ -133,30 +135,35 @@ type KernelVersion struct {
 	Minor int
 }
 
-// check verifies boot configuration on host.
-func (c *bootConfigParamChecker) check(ctx context.Context, reporter health.Reporter) error {
+// check verifies boot configuration on host. Returns collected health probes.
+// Skips checks if boot configuration is not available.
+func (c *bootConfigParamChecker) check() (probes health.Reporter, err error) {
```

Comment:
Have `check` accept the context and update the readers to be context-aware (i.e. `parseBootConfig` would check whether the context has expired).